pkg/updater/slim_updater.go

```diff
@@ -0,0 +1,137 @@
+package updater
+
+import (
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/urlfmt"
+	"github.com/stackrox/scanner/pkg/repo2cpe"
+	"github.com/stackrox/scanner/pkg/wellknowndirnames"
+)
+
+const (
+	repoToCPEFilename = "rhelv2/repository-to-cpe.json"
+)
+
+var (
+	slimUpdaterDir = filepath.Join(wellknowndirnames.WriteableDir, "slim-updater-artifacts.d")
+)
+
+// SlimUpdater updates the Scanner's definitions for scanner slim, contacting
+// Sensor, instead of Central as the Updater does.
+type SlimUpdater struct {
+	interval        time.Duration
+	lastUpdatedTime time.Time
+	stopSig         *concurrency.Signal
+
+	sensorClient           *http.Client
+	repoToCPE              *repo2cpe.Mapping
+	repoToCPELocalFilename string
+	repoToCPEURL           string
+}
+
+// NewSlimUpdater creates and initialize a new slim updater.
+func NewSlimUpdater(updaterConfig Config, sensorEndpoint string, repoToCPE *repo2cpe.Mapping) (*SlimUpdater, error) {
+	// Get the most recent genesis dump UUID, and construct the update URL.
+	uuid, err := getMostRecentGenesisDumpUUID()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting genesis UUID")
+	}
+	repoToCPEURL, err := urlfmt.FullyQualifiedURL(
+		strings.Join([]string{
+			urlfmt.FormatURL(sensorEndpoint, urlfmt.HTTPS, urlfmt.NoTrailingSlash),
+			"scanner/definitions",
+		}, "/"),
+		url.Values{
+			"uuid": []string{uuid},
+			"file": []string{repoToCPEFilename},
+		})
+	if err != nil {
+		return nil, errors.Wrapf(err, "setting up sensor URL at %s", sensorEndpoint)
+	}
+
+	// Create sensor's HTTP client.
+	sensorClient, err := clientconn.NewHTTPClient(
+		mtls.SensorSubject, urlfmt.FormatURL(sensorEndpoint, urlfmt.NONE, urlfmt.NoTrailingSlash), defaultTimeout)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating sensor client")
+	}
+
+	// Set up the repo2cpe local filename and its directory.
+	repoToCPELocalFilename := path.Join(slimUpdaterDir, repoToCPEFilename)
```

Comment:
nit: `filepath` should be used for filesystem ops

Reply:
Is there a need for us to be concerned about being dir-separator-agnostic?

Reply:
[filepath](https://pkg.go.dev/path/filepath) ensures the operations are compatible with the OS. Since we are opensource now, it's possible (for example) to someone to decide to build Scanner on a Windows machine, which would be incompatible with [path](https://pkg.go.dev/path)

Reply:
I mean, Scanner is a containerized app, and inside the container, we don't have to worry about the directory separator, right? This would also be true for building Scanner since it also happens in a container.